### PR TITLE
fix: #473 make concat compatible with (not)oneOf

### PR DESCRIPTION
**Pull request by Philipp91**

### CODE REVIEW

**Comment by Philipp91** [comment]
Bumping this. Is there anything missing for this PR?

**Comment by gregie156** [comment]
Yup team, please approve this. It's a MAJOR bug, and the fix is very professional.

**Comment by Philipp91** [comment]
@jquense explicitly (not sure how your GitHub notifications are set up)

File: src/mixed.js [hunk 1]
```diff
@@ -115,12 +131,21 @@ const proto = (SchemaType.prototype = {
...
     if (has(schema, '_default')) next._default = schema._default;
 
     next.tests = this.tests;
     next._exclusive = this._exclusive;
 
+    // manually merge the blacklist/whitelist (the other `schema` takes
+    // precedence in case of conflicts)
+    next._whitelist = this._whitelist
+      .union(schema._whitelist)
+      .setminus(schema._blacklist);
```
Comment (jquense) [inline src/mixed.js]: this is doing a bit more work than needed, lets roll `union` and `setminus` into a single function, e.g. `merge(newItems, removeItems)`

Comment (Philipp91) [reply src/mixed.js]: Done

**Comment by Philipp91** [comment]
@jquense I made the requested changes.

**Comment by jquense** [comment]
thanks!

**Comment by LKay** [comment]
When this will be released as a new version? That's pretty ugly bug which requires to copy-paste `oneOf` rules all over the place.

**Pull request merged**